src/util/fileupload.ts

```diff
@@ -90,4 +92,57 @@ export async function handleAvatarLinkAndGetURL(user, url) {
   return avatarUrl;
 }
 
+export async function uploadFileToBookCache({
+  path,
+  file,
+  contentType = 'image/png',
```

Comment:
I think making this required instead of optional is less error prone
```suggestion
  contentType,
```